### PR TITLE
Issue #1237 Don't show empty headers in release notes

### LINKED ISSUE #1237: Don't show empty headers in release notes
Hi,

Currently, if there is not issues related to tasks/bugs/features in milestone, then as output of `issue_list.sh` script it show the headers as empty as : 

```
# Features

# Bugs

# Tasks
```

### PR DESCRIPTION
**Pull request by budhrg**
Fix #1237 

## Examples

```
$ ./scripts/release/issue-list.sh -r minishift -m 23

# Bugs

- Issue [#1235](https://github.com/minishift/minishift/issues/1235) (bug) - Wrong version info in v1.4.0 binaries

$ ./scripts/release/issue-list.sh -r minishift -m 22
# Features

- Issue [#1133](https://github.com/minishift/minishift/issues/1133) (feature) - Function configContains in integration_test.go does not work properly

# Bugs

- Issue [#1121](https://github.com/minishift/minishift/issues/1121) (bug) - Possible error in delete/registration code
- Issue [#1123](https://github.com/minishift/minishift/issues/1123) (bug) - Instructions for manual installation of xhyve driver do not work on Sierra
- Issue [#1153](https://github.com/minishift/minishift/issues/1153) (bug) - Proxy not set for Docker engine
- Issue [#1161](https://github.com/minishift/minishift/issues/1161) (bug) - "developer document" on page "contribution Guidelines" show 404
- Issue [#1164](https://github.com/minishift/minishift/issues/1164) (bug) - Duplicate IDs in docs hamper downstreaming
- Issue [#1169](https://github.com/minishift/minishift/issues/1169) (bug) - Unsupported cluster up flags are not returned as error

# Tasks

- Issue [#1098](https://github.com/minishift/minishift/issues/1098) (task) - Update documentation: minishift.exe only works properly from C:""
- Issue [#1148](https://github.com/minishift/minishift/issues/1148) (task) - Reference the latest OpenShift doc in Minishift documentation
- Issue [#1152](https://github.com/minishift/minishift/issues/1152) (task) - Describe the issue of filling up the live root filesystem
- Issue [#1172](https://github.com/minishift/minishift/issues/1172) (task) - Fix Travis CI to run on Trusty
- Issue [#1177](https://github.com/minishift/minishift/issues/1177) (task) - Update to latest CentOS ISO

```

### CODE REVIEW

File: scripts/release/issue-list.sh [hunk 1]
```diff
@@ -23,6 +23,15 @@ function usage()
     exit 1
 }
 
+function discard_content_with_empty_issues() {
+  echo $1 | grep -q '[0-9]'
+  if [[ "$?" != 0 ]]; then
```
Comment (LalatenduMohanty) [inline scripts/release/issue-list.sh]: Personally I would have done this
```
no_of_lines=`echo -e "$1" | wc -l`
if [ ${no_of_lines} == "3" ]; then
```
Also to make the above to work you have to
```diff
-  features="# Features\n"
+  features="\n# Features\n"

```

File: scripts/release/issue-list.sh [hunk 2]
```diff
@@ -55,7 +64,11 @@ function milestone_issues()
     if [[ "$line" == *"(task)"* ]]; then tasks="$tasks\n$line"; fi
   done <<< "$issue_list"
 
-  echo -e "$features\n$bugs\n$tasks"
+  features=$(discard_content_with_empty_issues "$features")
+  bugs=$(discard_content_with_empty_issues "$bugs")
+  tasks=$(discard_content_with_empty_issues "$tasks")
+
+  echo -e "$features$bugs$tasks"
```
Comment (LalatenduMohanty) [inline scripts/release/issue-list.sh]: Also we can restructure the code. We can check each category and if it is not empty then we can directly print them to console them. If you want you can do that in a separate function .

File: scripts/release/issue-list.sh (hunk 1 above)
Comment (budhrg) [inline scripts/release/issue-list.sh]: > features="\n# Features\n"

That will add an unnecessary line at the beginning if features has more than 3 lines.

File: scripts/release/issue-list.sh (hunk 2 above)
Comment (budhrg) [inline scripts/release/issue-list.sh]: > Also we can restructure the code. We can check each category and if it is not empty then we can directly print them to console them. 

That part is done through `discard_content_with_empty_issues` function where I am emptying the variable `$bugs / $features / $ tasks`. The order is decided at the moment i.e Features -> Bugs -> Tasks.

Comment (budhrg) [inline scripts/release/issue-list.sh]: @LalatenduMohanty Fixed

File: scripts/release/issue-list.sh [hunk 3]
```diff
@@ -55,7 +62,9 @@ function milestone_issues()
     if [[ "$line" == *"(task)"* ]]; then tasks="$tasks\n$line"; fi
   done <<< "$issue_list"
 
-  echo -e "$features\n$bugs\n$tasks"
+  print_content_with_nonempty_issues "$features"
+  print_content_with_nonempty_issues "$bugs"
+  print_content_with_nonempty_issues "$tasks"
```
Comment (LalatenduMohanty) [inline scripts/release/issue-list.sh]: I think we should do this in a loop
for i in $features $bugs $tasks
do
    print_content_with_nonempty_issues $i
done

**Pull request merged**